flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSpeculativeExecutionSupport.java

```diff
@@ -144,9 +151,9 @@ public void testSpeculativeExecution() throws Exception {
   private static class TestingMap extends RichMapFunction<Row, Row> {
     @Override
     public Row map(Row row) throws Exception {
-      // Put the subtasks with the first attempt to sleep to trigger speculative
-      // execution
-      if (getRuntimeContext().getTaskInfo().getAttemptNumber() <= 0) {
+      // Simulate slow subtask 0 with attempt 0
+      TaskInfo taskInfo = getRuntimeContext().getTaskInfo();
+      if (taskInfo.getIndexOfThisSubtask() == 0 && taskInfo.getAttemptNumber() <= 0) {
```

Comment:
after this change of inferring source parallelism, this test would hang (even with the inferring parallelism flag turned off. It seems that speculative execution won't kick in somehow. This line of change seems to fix the problem however (tried local run 50 times without a failure). Not sure exactly why. Regardless of the reason, this seems like a good change anyway.

@pvary @venkata91 @becketqin let me know if you have any idea.

Reply:
Interesting. Thanks for tagging me. Do you mean adding `taskInfo.getIndexOfThisSubtask() == 0` solves the hang issue?

Reply:
@venkata91 that is correct.